library/modules/Units.cpp

```diff
@@ -2346,3 +2346,15 @@ void Units::setGroupActionTimers(color_ostream &out, df::unit *unit, int32_t amo
         }
     }
 }
+
+void Units::chopTree(df::unit *un, int16_t x, int16_t y, int16_t z)
+{
+    CHECK_NULL_POINTER(un);
+
+    auto fp = df::global::unitst_choptree;
+    CHECK_NULL_POINTER(fp);
```

Comment:
you might not want to throw here. 50.10, which is still supported in the 50.11 binaries, won't have this global defined. maybe check null and return without action if not set?

Reply:
i'd rather have it throw than fail silently. note also that this code is not currently called from any existing code as it's an extension to the API, not a reimplementation

Reply:
> For what it's worth, though, exceptions like the ones thrown by `CHECK_NULL_POINTER` will be converted to Lua errors if this function is called from Lua, so only C++ code would be at risk of crashing.

it occurs to me that we might be able to wrap a try block around plugin code and force-unload any plugin that throws instead of crashing DF as a whole, but that is a discussion for another day

for now, i'll have the function return a bool and it'll be up to downstream code to check for "success" that way. not happy about this really but for now it's probably our best available option